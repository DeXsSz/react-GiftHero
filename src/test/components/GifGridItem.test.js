import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../component/GifGridItem';

describe('Prueba en GifGridItem.jsx', () => {
    const title = 'Hola mundo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('debe de mostrar el componente GifGridItem correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de tener un parrafo con el titulo', () => {
        const parrafo = wrapper.find('p').text().trim();
        expect(parrafo).toBe(title);
    });
    test('debe una imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener la class de animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__fadeIn')).toEqual(true);
    });
});
