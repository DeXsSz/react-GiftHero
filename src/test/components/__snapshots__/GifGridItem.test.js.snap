// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGridItem.jsx debe de mostrar el componente GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <img
    alt="Hola mundo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Hola mundo
  </p>
</div>
`;
